### PR TITLE
Feature mes 4195 submit data to rsis

### PR DESCRIPTION
**Pull request by rstepnowski**
# Description and relevant Jira numbers

## Pull Request checklist

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

**Review by faggc — APPROVED** [review]

**Review by rossfauldsbjss — APPROVED** [review]
looks good

File: src/functions/uploadToRSIS/application/data-mapper/data-mapper.ts [hunk 1]
```diff
@@ -32,12 +34,20 @@ export const mapDataForMIExport = (result: ResultUpload): DataField[] => {
...
   // then map the data fields relevant to the specific test category
   const category = result.testResult.category;
-  if (category === 'B') {
-    mappedDataFields = mappedDataFields.concat(mapCatBData(result));
-  } else {
-    const message = `Unsupported Category: ${category}`;
-    error(message);
-    throw new Error(message);
+
+  switch (category) {
```
Comment (ammarhaiderbjss) [inline src/functions/uploadToRSIS/application/data-mapper/data-mapper.ts:38]: Would be good to have a unit test to cover this

File: src/functions/uploadToRSIS/application/data-mapper/data-mapper.ts [hunk 2]
```diff
@@ -157,6 +167,10 @@ export const formatQuestionFault = (testData: TestData | undefined): BooleanAsNu
   return 0;
 };
 
+export const formatQuestionFaultBE = (testData: TestData | undefined): number => {
```
Comment (ammarhaiderbjss) [inline src/functions/uploadToRSIS/application/data-mapper/data-mapper.ts:170]: Quick unit tests for these new methods too

File: src/functions/uploadToRSIS/application/data-mapper/__tests__/cat-be-mapper.spec.ts [hunk 3]
```diff
@@ -0,0 +1,1378 @@
...
+            bookingSequence: 11,
+            checkDigit: 3,
+          },
+        },
+        activityCode: '22',
+      },
+      autosaved: 0, // false
+    };
+
+    const expected: DataField[] = [
```
Comment (ammarhaiderbjss) [inline src/functions/uploadToRSIS/application/data-mapper/__tests__/cat-be-mapper.spec.ts:54]: this mock data should be extracted out to a .json file

**Review by ammarhaiderbjss — APPROVED** [review]
Happy to let it through provided we address the comments promptly

**Review by szabikr — COMMENTED** [review]
I think this needs a bit of a refactor. Which would require putting together a mock data file which would be referenced from those tests so that we don't have a spec file that is more than a 1000 lines

**Pull request merged**